packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -211,6 +211,9 @@ const AccountWizard = props => {
                 return { ...accumulator };
             }, {});
         }
+        if (shouldHideOccupationField(data?.employment_status)) {
+            delete data?.occupation;
+        }
```

Comment:
why do we delete occupation field value here?  Shouldnt we hide occcupation filed if shouldHideOccupationField is true. So this should be handled before this step..right?

Reply:
If u see the component structure, the form fields in `financial-details.tsx`  are not rendered based on form values. We just hide the field, when the form is rendered , bit not generate the form data object as per the fields. So before submission of data, I am removing the prop if the condition matches and if the `occupation` field is available

Reply:
we are already deletng this extra parameter in financial-assessment.tsx  -onsubmit function. Why do we do it agin?